### PR TITLE
Gg discovery

### PR DESCRIPTION
**Pull request by JonathanHenson**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: discovery/include/aws/discovery/GGGroup.h [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+#include <aws/discovery/GGCore.h>
+
+namespace Aws
+{
+    namespace Discovery
+    {
+        class AWS_DISCOVERY_API GGGroup final
+        {
+          public:
+            GGGroup() = default;
```
Comment (justinboswell) [inline discovery/include/aws/discovery/GGGroup.h]: They're GGGREAT!

I was going to suggest a namespace or the use of Greengrass, but then it turns out this is the nomenclature GG uses :\

File: discovery/source/DiscoveryClient.cpp [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+
+            Crt::StringStream ss;
+            ss << "greengrass-ats.iot." << clientConfig.region << ".amazonaws.com";
+
+            Crt::Io::TlsConnectionOptions tlsConnectionOptions = clientConfig.tlsContext->NewConnectionOptions();
+            uint16_t port = 443;
+
+            if (Crt::Io::TlsContextOptions::IsAlpnSupported())
+            {
+                tlsConnectionOptions.SetAlpnList("x-amzn-http-ca");
```
Comment (graebm) [inline discovery/source/DiscoveryClient.cpp]: Does this need follow-up changes in aws-c-http to work?

File: discovery/source/DiscoveryClient.cpp [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+
+                        requestOptions.headerArray = &hostNameHeader;
+                        requestOptions.headerArrayLength = 1;
+
+                        requestOptions.onStreamOutgoingBody = nullptr;
+                        requestOptions.onIncomingHeaders =
+                            [](Crt::Http::HttpStream &, const Crt::Http::HttpHeader *, std::size_t) {};
+                        requestOptions.onIncomingHeadersBlockDone =
+                            [callbackContext](Crt::Http::HttpStream &stream, bool) {
+                                callbackContext->responseCode = stream.GetResponseStatusCode();
```
Comment (graebm) [inline discovery/source/DiscoveryClient.cpp]: [optional] this callback could be null and you could just query the response-code in the onStreamComplete() callback

File: discovery/include/aws/discovery/ConnectivityInfo.h [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+* A copy of the License is located at
+*
+*  http://aws.amazon.com/apache2.0
+*
+* or in the "license" file accompanying this file. This file is distributed
+* on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+* express or implied. See the License for the specific language governing
+* permissions and limitations under the License.
+
+* This file is generated
```
[the remark this answers is not in the document]
Comment (JonathanHenson) [reply discovery/include/aws/discovery/ConnectivityInfo.h]: these aren't generated.

File: discovery/source/DiscoveryClient.cpp (hunk 2 above)
Comment (JonathanHenson) [reply discovery/source/DiscoveryClient.cpp]: yep, and done

File: discovery/source/DiscoveryClient.cpp (hunk 3 above)
Comment (JonathanHenson) [reply discovery/source/DiscoveryClient.cpp]: I wanted the ability to branch on it before writing to the stream. When we start generating this stuff I'll prolly revisit it.

**Pull request merged**